File: security/src/main/java/com/sreMake/security/spring/SecurityConf.java
package com.sreMake.security.spring;

import com.sreMake.common.conf.AppENV;

import java.util.List;
import java.util.stream.Stream;

public class SecurityConf {
    /**
     * 获取白名单列表
     */

    public static final List<String> AUTH_LIST = List.of(
            "/api/v1/auth/login",
            "/api/v1/auth/captcha",
            "/api/v1/auth/logout"
    );

    public static final List<String> ACTUATOR_LIST = List.of("/actuator/**");

    public static final List<String> JIMMER_OPENAPI = List.of(
            "/jimmer-client/**",
            "/openapi.html",
            "/openapi.yml"
    );

    public static final List<String> WHITE_LIST_WHIT_ENV = System.getenv("APP_ENV").equalsIgnoreCase(AppENV.PROD.name()) ? AUTH_LIST : Stream.concat(Stream.concat(AUTH_LIST.stream(), JIMMER_OPENAPI.stream()), ACTUATOR_LIST.stream()).toList();


}

File: security/src/main/kotlin/com/sreMake/security/spring/SecurityConf.kt
package com.sreMake.security.spring

import com.sreMake.common.conf.AppENV
import com.sreMake.common.utils.logger

object SecurityConf {
    /**
     * 获取白名单列表
     */

    @JvmStatic
    val authList: List<String> = listOf(
        "/api/v1/auth/login",
        "/api/v1/auth/captcha",
        "/api/v1/auth/logout",
        "/api/v1/auth/loginWithoutCaptcha"
    )


    @JvmStatic
    val actuatorList: List<String> = listOf("/actuator/**")


    @JvmStatic
    val jimmerOpenapi: List<String> = listOf(
        "/jimmer-client/**",
        "/openapi.html",
        "/openapi.yml",
        "/ts.zip"
    )

    @JvmStatic
    val whiteListWhitEnv: List<String> = System.getenv("APP_ENV")?.let {
        if (it.equals(AppENV.PROD.name, ignoreCase = true)) {
            mutableListOf<String>().apply {
                addAll(authList)
                addAll(actuatorList)
            }
        } else {
            mutableListOf<String>().apply {
                addAll(authList)
                addAll(jimmerOpenapi)
                addAll(actuatorList)
            }
        }
    } ?: authList.also {
        logger().warn("event 'APP_ENV' not set default use prod white list")
    }


}